### PR TITLE
Suggestion getmonsterinfo update

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: N/A

* **Server Mode**: both

* **Description of Pull Request**: 

Suggestion that gives `getmonsterinfo` the possibility to use mob id or monster name as argument.

### CODE REVIEW

File: src/map/script.cpp [hunk 1]
```diff
@@ -18069,21 +18069,23 @@ BUILDIN_FUNC(delmonsterdrop)
...
+	std::shared_ptr<s_mob_db> mob;
+
+	if (script_isstring(st, 2))
+		mob = mob_db.find(mobdb_searchname(script_getstr(st, 3)));
+	else
+		mob = mob_db.find(script_getnum(st, 2));
 
-	mob_id	= script_getnum(st,2);
-	if (!mobdb_checkid(mob_id)) {
+	if (mob == nullptr || mob_is_clone(mob->id)) {
```
Comment (Lemongrass3110) [inline src/map/script.cpp:18079]:
```suggestion
	if (mob == nullptr) {
```

File: doc/script_commands.txt [hunk 2]
```diff
@@ -3578,14 +3578,15 @@ This command does not count skills which are set as flag 4 (permament granted) (
 ---------------------------------------
 
 *getmonsterinfo(<mob ID>,<type>)
+*getmonsterinfo(<mob name>,<type>)
 
 This function will look up the monster with the specified ID number in the
```
Comment (Lemongrass3110) [inline doc/script_commands.txt:3583]:
```suggestion
This function will look up the monster with the specified <mob ID> or <mob name> in the
```

File: doc/script_commands.txt [hunk 3]
```diff
@@ -3578,14 +3578,15 @@ This command does not count skills which are set as flag 4 (permament granted) (
...
 mob database and return the info set by TYPE argument.
```
Comment (Lemongrass3110) [inline doc/script_commands.txt:3584]:
```suggestion
mob database and return the info set by <type> argument.
```

File: src/map/script.cpp [hunk 4]
```diff
@@ -18069,21 +18069,23 @@ BUILDIN_FUNC(delmonsterdrop)
  *------------------------------------------*/
 BUILDIN_FUNC(getmonsterinfo)
 {
-	int mob_id;
+	std::shared_ptr<s_mob_db> mob;
+
+	if (script_isstring(st, 2))
+		mob = mob_db.find(mobdb_searchname(script_getstr(st, 3)));
```
Comment (Lemongrass3110) [inline src/map/script.cpp:18075]:
```suggestion
		mob = mobdb_search_aegisname(script_getstr(st, 3));
```

File: src/map/script.cpp [hunk 5]
```diff
@@ -18069,21 +18069,23 @@ BUILDIN_FUNC(delmonsterdrop)
  *------------------------------------------*/
 BUILDIN_FUNC(getmonsterinfo)
 {
-	int mob_id;
+	std::shared_ptr<s_mob_db> mob;
```
Comment (Lemongrass3110) [inline src/map/script.cpp:18072]:
```suggestion
	std::shared_ptr<s_mob_db> mob = nullptr;
```

File: src/map/script.cpp [hunk 6]
```diff
@@ -18069,21 +18069,23 @@ BUILDIN_FUNC(delmonsterdrop)
...
+
+	if (script_isstring(st, 2))
+		mob = mob_db.find(mobdb_searchname(script_getstr(st, 3)));
+	else
+		mob = mob_db.find(script_getnum(st, 2));
```
Comment (Lemongrass3110) [inline src/map/script.cpp:18077]:
```suggestion
	else {
		uint16 mob_id = script_getnum(st, 2);

		if (!mob_is_clone(mob_id)) {
			mob = mob_db.find(mob_id);
		}
	}
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/script.cpp [hunk 7]
```diff
@@ -18066,24 +18066,32 @@ BUILDIN_FUNC(delmonsterdrop)
...
  *------------------------------------------*/
 BUILDIN_FUNC(getmonsterinfo)
 {
-	int mob_id;
+	std::shared_ptr<s_mob_db> mob = nullptr;
 
-	mob_id	= script_getnum(st,2);
-	if (!mobdb_checkid(mob_id)) {
+	if (script_isstring(st, 2))
+		mob = mob_db.find(mobdb_search_aegisname(script_getstr(st, 2)));
```
Comment (Lemongrass3110) [inline src/map/script.cpp:18076]:
```suggestion
		mob = mobdb_search_aegisname(script_getstr(st, 2));
```

**Review by Lemongrass3110 — APPROVED** [review]
Something strange happend during merging O_o
Or is it just github trolling me again?

File: src/map/script.cpp (hunk 7 above)
Comment (Atemo) [reply src/map/script.cpp]: Maybe I made a mistake !

**Pull request merged**